File: Problem-3.js
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : None


// Your code here along with comments explaining your approach

/**
 * @param {string} s
 * @return {number}
 */
 var longestPalindrome = function(s) {
    let frequency = {};
    
    s.split('').forEach((item) => {
        frequency[item] = frequency[item] ? frequency[item] + 1 : 1;
    });
    
    let oddCount = 0;
    let evenCount = 0;
    
    for (let item in frequency) {
        if (frequency[item] % 2 === 0) {
            evenCount += frequency[item];
        } else {
            evenCount += frequency[item] - 1;
            oddCount = 1;
        }
    }
    return evenCount + oddCount;
};

File: problem_3.cpp
/*

// Time Complexity : O(N) length of the string
// Space Complexity : O(1) constant cause fixed arr length to store hash set
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this :


// Your code here along with comments explaining your approach

*/

#include<vector>
#include<string>

using namespace std;

class Solution {
    const int vec_len{123-65};
public:
    int longestPalindrome(string s) {
        int count{},pal_long{};
        vector<bool> hash_set(vec_len,false);
        for(auto ch:s){
            int key = ch-65;
            // storing the ocurrence of the alphabet
            if(hash_set.at(key) == false){
                hash_set.at(key) = true;
                ++count;
            }
            // if occured add two to the pal_long cause 2 characters and remove the alphabet
            else {
                pal_long+=2;
                hash_set.at(key)=false;
                --count;
            }
        }// check if some remainig so any single element can be held in middle for the odd palindrome,
        if(count>0){
            ++pal_long;
        }
        return pal_long;
    }
};